BranchSDK/src/BranchIO/Event/SessionEvent.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019-21 Branch Metrics, Inc.
+
+#include "BranchIO/Event/SessionEvent.h"
+#include "BranchIO/Util/Log.h"
+
+#include <Poco/URI.h>
+
+#include <sstream>
+
+using namespace std;
+using namespace Poco;
+
+namespace BranchIO {
+
+BaseEvent& SessionOpenEvent::setLinkUrl(const String &url) {
+    string surl(url.str());
+
+    // The link_identifier is sent in JSON as a number, equal
+    // to the link_click_id query parameter from the inbound
+    // URI (url argument here).
+    try {
+        URI uri(surl); // throws SyntaxException when invalid
+
+        // vector<pair<string, string>>
+        // ordered, not associative container
+	auto queryParams = uri.getQueryParameters();
```

Comment:
Here's the meat of the change. This method uses Poco to get the query parameters from the inbound URI and looks for `link_click_id`.